### PR TITLE
added external api service OpenLibraryService

### PR DESCRIPTION
**Pull request by mikolaslorant**
## Summary

* added endpoint to get bookk by ISBN.
* added OpenLibraryService that uses external book api.
* added AuthorDTO, PublisherDTO and BookInfoDTO

## Trello card

https://trello.com/c/fP99JBMN/22-integraci%C3%B3n-api-externa

### CODE REVIEW

**Review by IgnOliveto — APPROVED** [review]

**Review by IgnOliveto — DISMISSED** [review]

File: training/src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -68,6 +69,18 @@ public Book() {
...
+        this.users = new HashSet<>();
+        this.title = bookInfo.getTitle();
+        this.subtitle = bookInfo.getSubtitle();
+        this.year = bookInfo.getPublishDate();
+        this.isbn = bookInfo.getIsbn();
+        this.author = bookInfo.getAuthors()[0].getName();
+        this.publisher = bookInfo.getPublishers()[0].getName();
+        this.genre = "default_genre";
+        this.pages = bookInfo.getPages();
+        this.image = "default_image";
```
Comment (leandrocostawolox) [inline training/src/main/java/wolox/training/models/Book.java]: The openlibrary API provides an image, but it's given with the name of "cover"

File: training/src/main/java/wolox/training/services/OpenLibraryService.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode root = null;
+        try {
+            root = mapper.readTree(response.getBody());
+        } catch (IOException e) {
+            throw new BookNotFoundException();
+        }
+        BookInfoDTO bookInfoDTO = mapper.convertValue(root.path("ISBN:" +isbn), BookInfoDTO.class);
+        bookInfoDTO.setIsbn(isbn);
+        bookInfoDTO.setSubtitle(DEFAULT_SUBTITLE);
```
Comment (leandrocostawolox) [inline training/src/main/java/wolox/training/services/OpenLibraryService.java]: The openlibrary API provides a subtitle

**Review by leandrocostawolox — CHANGES_REQUESTED** [review]
Good job! The only thing to comment: There are two fields which are not mapped from the open library API

File: training/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,6 @@
+#Sat Aug 17 12:43:15 ART 2019
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
```
Comment (mikolaslorant) [inline training/gradle/wrapper/gradle-wrapper.properties:2]: This was the configuration that was giving me a headache. Don't know why it wasn't set up like this in the first place.

**Review by leandrocostawolox — APPROVED** [review]

**Review by IgnOliveto — APPROVED** [review]

**Pull request merged**